### PR TITLE
docs(proxy): Add `README.md`

### PR DESCRIPTION
**Pull request by duesee**
To be merged after #51

### CODE REVIEW

File: proxy/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+The `security` field configures TLS. `Insecure` disables TLS encryption and generally SHOULD NOT be used when proxying
+to a remote server.
+
+## Features
+
+* Async
+* Abstracts away literal handling
+* Fully supports unsolicited responses
```
Comment (jakoschiko) [inline proxy/README.md:58]: Do you really describe the proxy here or `imap-flow`?

File: proxy/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# IMAP Proxy
+
+Proxy intercepting IMAP connections and `Debug`-printing them in the terminal.
```
[the remark this answers is not in the document]
Comment (jakoschiko) [reply proxy/README.md:3]: Wasn't logging disabled by default? Then we would need to document how this printing can be enabled.

[the remark this answers is not in the document]
Comment (duesee) [reply proxy/README.md:3]: > Wasn't logging disabled by default? Then we would need to document how this printing can be enabled.

Oh, good point! I *knew* [direnv](https://direnv.net/) would bite me at some point :P

File: proxy/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+```text
+Usage: proxy [--config <config>]
+
+IMAP proxy.
+
+Options:
+  --config          optional config path ("config.toml" by default)
+  --help            display usage information
+```
```
[the remark this answers is not in the document]
Comment (duesee) [reply proxy/README.md:38]: Let's remove it 👍🏻

File: proxy/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (duesee) [reply proxy/README.md]: I added `RUST_LOG=proxy=trace`. I think it's okay for now.

File: proxy/README.md [hunk 4]
```diff
@@ -48,34 +38,34 @@ bind = { host = "127.0.0.1", port = 1143, security = "Insecure" }
 connect = { host = "imap.example.org", port = 993 }
 ```
 
-The `security` field configures TLS. `Insecure` disables TLS encryption and generally SHOULD NOT be used when proxying
-to a remote server.
+The `security` field configures TLS. `Insecure` disables TLS encryption and generally SHOULD NOT be used when proxying to a remote server.
```
Comment (duesee) [inline proxy/README.md:41]: Only whitespace, sorry for the churn.

File: proxy/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakoschiko) [reply proxy/README.md]: I'm not sure, it's getting weirder. You added it to the line that calls the help text and doesn't do any logging. Maybe a little bit more verbosity would be beneficial here.

File: proxy/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+# IMAP Proxy
+
+A proxy that receives, `Debug`-prints, and forwards IMAP messages.
+
+It should™ forward all messages *unchanged*. Still, ...
```
Comment (jakoschiko) [inline proxy/README.md:5]: Technically this is not true by design. E.g. see `ServerFlowOptions::literal_accept_text` which will produce messages never sent by the real server.

Comment (duesee) [reply proxy/README.md:5]: Grrr... does a footnote suffice? *Technically*, it's also not true on the syntax level as the IMAP syntax is not canonical, e.g., `a login` == `a LOGIN`. Important part is: It doesn't change the semantics.

**Review by jakoschiko — APPROVED** [review]

**Pull request merged**